components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -383,9 +399,20 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             error!(self.logger, "peer id not matched"; "persisted_peer_id" => peer_id, "persisted_number" => ready_number);
             return;
         }
-        let persisted_message = self
-            .async_writer
-            .on_persisted(ctx, ready_number, &self.logger);
+        let (persisted_message, has_snapshot) =
+            self.async_writer
+                .on_persisted(ctx, ready_number, &self.logger);
+        if has_snapshot {
```

Comment:
Only when `persisted_index == first_index && has_snapshot` should it assume the snapshot is applied.

Reply:
the first_index is the truncate_index+1 and the persisted_index is the truncate_index , so the condition should be persisted_index == first_index-1